main/cli_parser.cpp

```diff
@@ -0,0 +1,633 @@
+	add_command(help_command);
+
+	CommandFlag render_thread_command("--render-thread");
+	render_thread_command.set_data("Render thread mode ('unsafe', 'safe', 'separate').", "Run options");
+	render_thread_command.set_argument_name("<mode>");
+	render_thread_command.add_checker(check_render_thread, "Must be a valid render-thread");
+	add_command(help_command);
+
+	CommandFlag remote_fs_command("--remote-fs");
+	remote_fs_command.set_data("Remote filesystem (<host/IP>[:<port>] address).", "Run options");
+	remote_fs_command.set_argument_name("<address>");
+	add_command(help_command);
+
+	CommandFlag remote_fs_password_command("--remote-fs-password");
+	remote_fs_password_command.set_data("Password for remote filesystem.", "Run options");
+	remote_fs_password_command.set_argument_name("<password>");
+	add_command(help_command);
+
+	CommandFlag audio_driver_command("--audio-driver");
+	String audio_driver_description = "Audio driver (";
+	for (int i = 0; i < OS::get_singleton()->get_audio_driver_count(); i++) {
+		if (i != 0)
+			audio_driver_description += ", ";
+		audio_driver_description += OS::get_singleton()->get_audio_driver_name(i);
+	}
+	audio_driver_description += ").";
+	audio_driver_command.set_data(audio_driver_description, "Run options");
+	audio_driver_command.add_checker(check_audio_driver, "Invalid audio driver");
+	add_command(help_command);
+
+	CommandFlag video_driver_command("--video-driver");
+	String video_driver_description = "Video driver (";
+	for (int i = 0; i < OS::get_singleton()->get_video_driver_count(); i++) {
+		if (i != 0)
+			video_driver_description += ", ";
+		video_driver_description += OS::get_singleton()->get_video_driver_name(i);
+	}
+	video_driver_description += ").";
+	video_driver_command.set_data(video_driver_description, "Run options");
+	video_driver_command.add_checker(check_video_driver, "Invalid video driver");
+	add_command(help_command);
+
+	// Display options
+	CommandFlag fullscreen_command("--fullscreen", "-f");
+	fullscreen_command.set_data("Request fullscreen mode.", "Display options");
+	add_command(help_command);
+
+	CommandFlag maximized_command("--maximized", "-m");
+	maximized_command.set_data("Request a maximized window.", "Display options");
+	add_command(help_command);
+
+	CommandFlag windowed_command("--windowed", "-w");
+	windowed_command.set_data("Request windowed mode.", "Display options");
+	add_command(help_command);
+
+	CommandFlag always_on_top_command("--always-on-top", "-t");
+	always_on_top_command.set_data("Request an always-on-top window.", "Display options");
+	add_command(help_command);
+
+	CommandFlag resolution_command("--resolution");
+	resolution_command.set_data("Request window resolution.", "Display options");
+	resolution_command.set_argument_name("<W>x<H>");
+	resolution_command.add_checker(check_resolution, "Invalid resolution, it should be e.g. '1280x720' and the values must be above 0.");
+	add_command(help_command);
```

Comment:
All the previous `add_command` calls are passing `help_command`, I guess by mistake.